javascript/javascript3/week2/promise.js

```diff
@@ -0,0 +1,93 @@
+ // Promise that resolves after set time
+
+function getResolveIn(resolveAfter) {
+   return new Promise(function(resolve, reject){
+       setTimeout(function() {
+           resolve(resolveAfter);
+
+       }, 1000 * resolveAfter);
+   });
+}
+getResolveIn(8).then(function() {
+   console.log("I am called asynchronously")
+});
+
+
+// Using the above promise with async/await
+
+async function getResolveIn(resolveAfter) {
+    let promise = new Promise((resolve, reject) => {
+        setTimeout(() => resolve("Calling after 8 seconds"), 1000 * resolveAfter)
+    });
+    let result = await promise;          // wait until the promise resolves.
+    console.log(result);
+}
+getResolveIn(4);
+
+
+
+  // Rewrite setTimeout and getCurrentPosition using geolocation
+
+function reWrite() {
+    return new Promise(function(resolve, reject) {
+        setTimeout(resolve, 3000);
+    }).then(function() {
+        console.log("Called after 3 seconds");
+    });
+}
+reWrite();
```

Comment:
I think it's a better way!!!

function reWrite(milliseconds) {
    return new Promise(function(resolve, reject) {
        setTimeout(function() {
            resolve();
        }, milliseconds);
    });
}

reWrite(3000).then(() => {
    console.log("Called after 3 seconds");
});